ingesters/HttpIngester/hec.go

```diff
@@ -105,16 +119,26 @@ func (hh *hecHandler) getDefaultTag(h *handler, r *http.Request, ll *log.KVLogge
 				return
 			} else {
 				ok = true
+				override = tagOverride{
+					param: parameterTag,
+					value: val,
+				}
 			}
 		} else if val = v.Get(parameterSourcetype); val != `` && len(hh.tagRouter) > 0 {
-			tg, ok = hh.tagRouter[val]
+			if tg, ok = hh.tagRouter[val]; ok {
+				override = tagOverride{
+					param: parameterSourcetype,
+					value: val,
```

Comment:
so here's we're returning the value of the sourcetype parameter, not the actual tag we looked up. I'm fine with that if it's what you want, just pointing out that the string we put here is *not* necessarily the tag where the data will go

Reply:
The goal is to log what was provided on the URL for debugging, not necessarily what it will be translated into.

I agree the naming isn't awesome, but calling it "tagSourcetypeWhateverOverride" feels bad?

Reply:
works for me